server/utils/utils.ts

```diff
@@ -307,14 +307,44 @@ export const utcTimestampToUtcDate = (str: string) => (str ? formatDate(new Date
 export const utcTimestampToUtcDateTime = (str: string) =>
   str ? formatDate(new Date(str), 'dd-MMM-yyyy HH:mm:ss').toUpperCase() : undefined
 
-export const createStrapiQuery = (fields: string[]): string => {
-  if (!fields || fields?.length === 0) {
-    return ''
+export function createStrapiQuery({ populate }: { populate?: string[] }): string {
+  const populateParams: Record<string, unknown> = {}
+
+  populate?.sort((a, b) => b.split('.').length - a.split('.').length)
+
+  populate?.forEach(path => {
+    const keys = path.split('.')
+    let current = populateParams
+
+    keys.forEach((key, index) => {
+      if (!current[key]) {
+        // Ensure the last key is set to true, and intermediate keys have a `populate` object
+        current[key] = index === keys.length - 1 ? true : { populate: {} as Record<string, unknown> }
+      }
+      current =
+        typeof current[key] === 'object' && current[key] !== null
+          ? (current[key] as { populate?: Record<string, unknown> }).populate ||
+            (current[key] as Record<string, unknown>)
+          : current
+    })
+  })
+
+  function buildQuery(
```

Comment:
To start with, think it would be good to extract this out of this function.. being nested in the other function makes this much more difficult to read as it implies that you're accessing some of the outer variables (which if you are is a bit of a smell in its own right)